src/vcpkg/registries.cpp

```diff
@@ -1143,22 +1143,20 @@ namespace vcpkg
         if (auto builtin_registry = dynamic_cast<BuiltinRegistry*>(default_registry_.get()))
         {
             default_registry_is_builtin = true;
-            builtin_registry->m_baseline_identifier.assign(baseline.begin(), baseline.end());
-        }
-
-        if (!default_registry_is_builtin || registries_.size() != 0)
-        {
-            System::print2(System::Color::warning,
-                           "Warning: when using the registries feature, one should not use `\"builtin-baseline\"` "
-                           "to set the baseline.\n",
-                           "    Instead, use the \"baseline\" field of the registry.\n");
+            if (builtin_registry->m_baseline_identifier.empty())
+            {
+                builtin_registry->m_baseline_identifier.assign(baseline.begin(), baseline.end());
```

Comment:
We should warn if `m_baseline_identifier.empty() && !baseline.empty()`

Reply:
I'm not sure about this; for example:

vcpkg.json:
```json
{
  ...,
  "builtin-baseline": "b1"
```

vcpkg-configuration.json:
```
{
  "registries": [
    {
      "kind": "builtin",
      "baseline": "b2",
      "packages": [ "a", "b", "c" ]
    }
  ]
}
```

Reply:
Right; so in this case we should warn to the user about which of the two baselines they've specified will be used (I assume b2?)

Reply:
I mean, they'll both be used: `b2` will be used for packages `a`, `b`, and `c`, and `b1` will be used for everything else.

Reply:
I thought that having `"packages": [ "a", "b", "c" ]` would disable all other packages from the built-in registry. 

If not, then trying to install package `d` when there is no `"builtin-baseline": "b1"` and also attempting to use version constraints will fail and we should warn about that.

Reply:
You can disable the builtin registry, but you must explicitly do so by setting `default-registry` to `null`.

Reply:
What is the syntax for changing the default registry's baseline? I suspect this means we should warn in the case that the user has specified `"builtin-baseline"` AND specifies `"default-registry"`. If that is already the case in this PR, then this comment is resolved.

Reply:
The syntax for changing the default registry's baseline is:

```json
{
  "default-registry": {
    "kind": "...",
    ...,
    "baseline": "<baseline>"
  }
}
```

This is not warned against in this PR; what should the warning look like, do you think?

Reply:
"Warning: Specifying the "default-registry" overrides the "builtin-baseline" field. Pass <blah> to suppress this warning message."

Reply:
Do we only want `builtin-baseline` to do something to a registry that is _both_ builtin, and default? Currently it does something to registries that are builtin, even if they're non-default.

Reply:
It should only affect the default builtin baseline in my opinion. I see the following scenarios:
1. No `vcpkg-configuration.json` (no warn)
2. adds other registries but doesn't change the default (no warn)
3. doesn't change the default, but _does_ re-add the builtin registry to get packages with a different baseline (no warn)
4. re-defines the default to builtin, but at a different baseline (warn)

I think the most straightforward UX is _all_ registries in the configuration file _must_ have a baseline specified. Therefore, the only case that `builtin-baseline` could possibly apply would be in the unspecified default case.